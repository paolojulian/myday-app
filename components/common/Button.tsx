import ThemedText from '@/components/common/ThemedText';
import ThemedView from '@/components/common/ThemedView';
import { colors } from '@/constants/Colors';
import React, { useMemo } from 'react';
import { StyleSheet } from 'react-native';
import { TouchableOpacity, TouchableOpacityProps } from 'react-native';

type ButtonVariants = 'primary';

type ButtonProps = {
  text: string;
  variant?: ButtonVariants;
} & TouchableOpacityProps;

function Button({ text, variant = 'primary', ...props }: ButtonProps) {
  const containerVariantStyle = useMemo(() => {
    switch (variant) {
      case 'primary':
        return styles.primaryContainer;
      default:
        return {};
    }
  }, [variant]);

  const textVariantStyle = useMemo(() => {
    switch (variant) {
      case 'primary':
        return styles.primaryText;
      default:
        return {};
    }
  }, [variant]);

  return (
    <TouchableOpacity activeOpacity={0.9} {...props}>
      <ThemedView style={[styles.container, containerVariantStyle]}>
        <ThemedText style={[textVariantStyle]}>{text}</ThemedText>
      </ThemedView>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  container: {
    borderRadius: 100,
    padding: 16,
    alignItems: 'center',
    justifyContent: 'center',
  },
  primaryContainer: {
    backgroundColor: colors.black,
  },
  primaryText: {
    color: colors.white,
  },
});

export default Button;
