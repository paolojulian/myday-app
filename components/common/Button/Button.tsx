import ThemedText from '@/components/common/ThemedText';
import ThemedView from '@/components/common/ThemedView';
import { colors } from '@/constants/Colors';
import React, { useMemo } from 'react';
import { StyleSheet } from 'react-native';
import { TouchableOpacity, TouchableOpacityProps } from 'react-native';

type ButtonVariants = 'primary';

type ButtonProps = {
  text: string;
  variant?: ButtonVariants;
} & TouchableOpacityProps;

function Button({ text, variant = 'primary', ...props }: ButtonProps) {
  const containerVariantStyle = useMemo(() => {
    switch (variant) {
      case 'primary':
        return styles.primaryContainer;
      default:
        return {};
    }
  }, [variant]);

  const textVariantStyle = useMemo(() => {
    switch (variant) {
      case 'primary':
        return styles.primaryText;
      default:
        return {};
    }
  }, [variant]);

  return (
    <TouchableOpacity role="button" accessibilityRole="button" activeOpacity={0.9} {...props}>
      <ThemedView style={[styles.container, containerVariantStyle]}>
        <ThemedText variant="header-sm" style={[textVariantStyle, styles.text]}>
          {text}
        </ThemedText>
      </ThemedView>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  container: {
    borderRadius: 100,
    padding: 16,
    alignItems: 'center',
    justifyContent: 'center',
  },
  primaryContainer: {
    backgroundColor: colors.v2.teal,
  },
  primaryText: {
    color: colors.v2.black,
  },
  text: {
    fontWeight: 'bold',
  },
});

export default Button;
