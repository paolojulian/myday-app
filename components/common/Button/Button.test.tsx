import Button from '@/components/common/Button/Button';
import { render, screen } from '@testing-library/react-native';

describe('TESTING Button Component', () => {
  describe('GIVEN a text', () => {
    const text = 'Save';
    describe('WHEN the component is rendered', () => {
      it('THEN it should match the snapshot', () => {
        const component = render(<Button text="Save" />);
        expect(component).toMatchSnapshot();
      });
      it('THEN it should render the given text', () => {
        render(<Button text={text} />);
        expect(screen.getByText(text)).toBeTruthy();
      });
    });
  });
});
