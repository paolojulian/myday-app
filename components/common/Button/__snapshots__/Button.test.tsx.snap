// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TESTING Button Component GIVEN a text WHEN the component is rendered THEN it should match the snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "borderRadius": 100,
          "justifyContent": "center",
          "padding": 16,
        },
        {
          "backgroundColor": "#05141F",
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "fontFamily": "Open_SansMedium",
            "fontSize": 16,
            "fontWeight": 500,
          },
          {
            "fontWeight": 400,
          },
          [
            {
              "color": "#FFFFFF",
            },
            {
              "fontWeight": "bold",
            },
          ],
        ]
      }
    >
      Save
    </Text>
  </View>
</View>
`;
